### PR TITLE
Some minor cleanup

### PR DESCRIPTION
**Pull request by fingolfin**
This is a small subset of the changes in #791 in particular it includes all the "manual" changes I made there. I hope this is less controversial and can just be merged.

(While here, may I ask if we could get a new juliaup release soon? I am eager to try out the new forking code, and verify that it fixes `ctrl-Z` for me ;-) )

### CODE REVIEW

File: src/operations.rs
Comment (LilithHafner) [inline src/operations.rs:162]: I don't see any particular reason to transfer ownership here if a reference works just as well, but still happy to merge this part of the diff if you think it's better for stylistic reasons.

**Review by LilithHafner — APPROVED** [review]
LGTM, My one comment is optional, feel free to accept or decline.

I would merge, but I'm not sure of a repo-wide merge-block is in effect @davidanthoff

File: src/operations.rs
Comment (davidanthoff) [reply src/operations.rs:162]: I agree, I actually prefer a ref here. But also, it really doesn't matter very much, so I'll merge this as soon as the merge window opens again.

Comment (fingolfin) [reply src/operations.rs:162]: Just to explain: I merely wanted the two copies of code calling `unpack_sans_parent` to match as closely as possible, so that future edits don't have to wonder "why are these different?". Since I had to decide with two options, I used what `cargo clippy` considers as "best practice".

**Pull request merged**
